src/components/widget-creator.jsx

```diff
@@ -107,100 +120,93 @@ const WidgetCreator = ({instId, widgetId, minHeight='', minWidth=''}) => {
 			}
 		}
 	})
-	
-	// qset storage for previous save feature
-	// current working qset, temporarily cached to await confirm/cancel
-	const [qsetToBeCached, setQsetToBeCached] = useState(null);
-	// qset selected to be loaded after requested reload
-	const [qsetToReload, setQsetToReload] = useState(null);
-	const [keepQSet, setKeepQSet] = useState(null);
-	const [invalid, setInvalid] = useState(null);
 
-	const { isLoading: qSetIsLoading, data: qset } = useQuery({
-		queryKey: ['qset', instId],
-		queryFn: () => apiGetQuestionSet(instId),
-		staleTime: Infinity,
-		placeholderData: null,
-		enabled: !!instId,
-		onSettled: (data) => {
-			if ( (data != null ? data.title : undefined) === 'Permission Denied' ||data.title === 'error') {
-					setInvalid(true);
-					onInitFail('Permission Denied')
-				} else {
-					setInvalid(false);
-					setKeepQSet(data);
-				}
-		}
-	})
-
-	// Initializes the instance after query is complete
+	// Calls embedCreator() once the widgetInstance and/or widgetInfo have loaded
 	useEffect(() => {
-
-		if (widgetInstance && Object.keys(widgetInstance).length !== 0) {
-			setInstance(widgetInstance)
-			setInstanceId(widgetInstance.id)
+		if (widgetInstance && widgetInstance.widget && Object.keys(widgetInstance.widget).length > 0) {
+			if (!widgetInstance.is_draft) {
+				setPublishText('Update')
+				setUpdateMode(true)
+			}
+			embedCreator(widgetInstance.widget);
 		}
-
-		if (widgetInfo && Object.keys(widgetInfo).length !== 0){
-			setInstance({...instance, widget: widgetInfo})
+		else if (widgetInfo) {
+			embedCreator(widgetInfo);
 		}
-	}, [instanceIsLoading, widgetInfoIsLoading])
 
-	// Calls embedCreator() once the instance and qset (if applicable) have loaded
-	useEffect(() => {
+		// Show the buttons that interact with the creator
+		enableReturnLink()
+		setShowActionBar(true)
 
-		if (instance.widget && Object.keys(instance.widget).length > 0) {
+	}, [widgetInstance, widgetInfo])
 
-			if ((instId && instance.hasOwnProperty('id'))) {
-				waitForQSet().then(() => {
-					embedCreator();
-				})
+	// Called only if instId exists and the fetch for qset was successful
+	useEffect(() => {
+		if (creatorReady) {
+			if (widgetInstance && (!keepQSet || Object.keys(keepQSet).length == 0)) {
+				return;
+			} else if (widgetInstance && frameRef.current == null) {
+				onInitFail('Unable to load widget.')
+			} else if (widgetInstance && !canPublish && !widgetInstance.is_draft) {
+				onInitFail('Widget type can not be edited by students after publishing.')
 			} else {
-				embedCreator();
-			}
+				if (widgetInstance && (keepQSet || qsetToReload)) {
```

Comment:
Just with functional testing the changes, I'm noticing that there's still an occasional hiccup where this condition is reached and resolves as false even for existing widgets. As best I can tell, since this entire check is wrapped in a `useEffect` that's set to fire on both changes to `creatorReady` and `keepQSet`, there are some situations where `keepQSet` has not populated when `creatorReady` is set to `true`, and when it finally updates, `creatorReady` has flipped back to false. I'm not 100% sure on a resolution other than changing when and how `setCreatorReady(false)` is called.